src/ipa/rpi/controller/rpi/awb_nn.cpp

```diff
@@ -0,0 +1,790 @@
+#include "awb_nn.h"
+
+#include <libcamera/base/file.h>
+#include <libcamera/base/log.h>
+
+#include <tensorflow/lite/kernels/register.h>
+
+#include "alsc_status.h"
+
+using namespace RPiController;
+using namespace libcamera;
+
+LOG_DECLARE_CATEGORY(RPiAwb)
+
+constexpr double kDefaultCT = 4500.0;
+
+#define NAME "rpi.nn.awb"
+
+static int readCtCurve(ipa::Pwl &ctR, ipa::Pwl &ctB, const libcamera::YamlObject &params)
+{
```

Comment:
I wonder if we could put this function into awb_base.h (or maybe we'll need awb_base.cpp) so that it can be shared?

Reply:
If we don't derive NN or Bayesian files from the base class, perhaps call this awb_common.*?